sympy/sets/sets.py

```diff
@@ -765,7 +763,9 @@ def __new__(cls, start, end, left_open=False, right_open=False):
         if end == S.Infinity:
             right_open = true
 
-        return Basic.__new__(cls, start, end, left_open, right_open)
+        interval = Basic.__new__(cls, start, end, left_open, right_open)
+        interval.is_EmptySet = (end - start).is_negative
```

Comment:
`if end == start and (left_open or right_open):
             return S.EmptySet`
Do you mean this case? I don't miss that.